mmocr/models/ner/utils/bert.py

```diff
@@ -0,0 +1,488 @@
+        output_attentions (bool):  Whether use the attentions in output
+        output_hidden_states (bool): Whether use the hidden_states in output.
+        num_attention_heads (int): The number of attention heads.
+        attention_probs_dropout_prob (float): The dropout probability
+            of attention.
+        intermediate_size (int): The size of intermediate layer.
+        hidden_act (str):  hidden layer activation
+    """
+
+    def __init__(self,
+                 num_hidden_layers=12,
+                 initializer_range=0.02,
+                 vocab_size=21128,
+                 hidden_size=768,
+                 max_position_embeddings=128,
+                 type_vocab_size=2,
+                 layer_norm_eps=1e-12,
+                 hidden_dropout_prob=0.1,
+                 output_attentions=False,
+                 output_hidden_states=False,
+                 num_attention_heads=12,
+                 attention_probs_dropout_prob=0.1,
+                 intermediate_size=3072,
+                 hidden_act='gelu_new'):
+        super().__init__()
+        self.embeddings = BertEmbeddings(
+            vocab_size=vocab_size,
+            hidden_size=hidden_size,
+            max_position_embeddings=max_position_embeddings,
+            type_vocab_size=type_vocab_size,
+            layer_norm_eps=layer_norm_eps,
+            hidden_dropout_prob=hidden_dropout_prob)
+        self.encoder = BertEncoder(
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            num_hidden_layers=num_hidden_layers,
+            hidden_size=hidden_size,
+            num_attention_heads=num_attention_heads,
+            attention_probs_dropout_prob=attention_probs_dropout_prob,
+            layer_norm_eps=layer_norm_eps,
+            hidden_dropout_prob=hidden_dropout_prob,
+            intermediate_size=intermediate_size,
+            hidden_act=hidden_act)
+        self.pooler = BertPooler(hidden_size=hidden_size)
+        self.num_hidden_layers = num_hidden_layers
+        self.initializer_range = initializer_range
+        self.init_weights()
+
+    def _resize_token_embeddings(self, new_num_tokens):
+        old_embeddings = self.embeddings.word_embeddings
+        new_embeddings = self._get_resized_embeddings(old_embeddings,
+                                                      new_num_tokens)
+        self.embeddings.word_embeddings = new_embeddings
+        return self.embeddings.word_embeddings
+
+    def forward(self,
+                input_ids,
+                attention_masks=None,
+                token_type_ids=None,
+                position_ids=None,
+                head_mask=None):
+        if attention_masks is None:
+            attention_masks = torch.ones_like(input_ids)
+        if token_type_ids is None:
+            token_type_ids = torch.zeros_like(input_ids)
+        extended_attention_mask = attention_masks.unsqueeze(1).unsqueeze(2)
```

Comment:
Since `attention_masks` is not used in this function below, you can rename `extended_attention_mask` to `attention_mask`